### PR TITLE
#551 FuncOf returning null on apply

### PR DESCRIPTION
**Pull request by SharplEr**
For #551 
**Problem**
`FuncOf` can wrap `Proc` without result value. And in such case `null` result was an expected behavior.
But `null` is prohibited by the design principles.

**Solution**
- I have removed ctors from `FuncOf` without result value, and add ctors with result value in connected classes (as `BiFuncOf`, etc).
- Also, replaced in `RunnableOf` usage of `Func` with `Proc`.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#843](https://github.com/yegor256/cactoos/issues/843) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#843](https://github.com/yegor256/cactoos/issues/843) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/C63314D6Z/1282107946); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by Vatavuk** [comment]
@SharplEr This PR is too big. First of all `SyncProc` should be issue for itself since it is an enhancment and doesn't directly solves this issue. Secondly there are 11 more classes with accompanied tests. I propose to leave original ctor with null in `FuncOf` and introduce one with `T result` just like you did in this PR. Then choose 3 or 4 classes and refactor them to use this newly created ctor. Create puzzles for the remaining classes.

**Comment by llorllale** [comment]
@SharplEr there was a problem generating the codecov report. Can you rebase your branch to 57a3c18175ef35e79a1bd1c35284be7db1c228b5 ?

**Comment by SharplEr** [comment]
@llorllale I just rebased with the last master. Don't see any problem.

**Comment by llorllale** [comment]
@SharplEr thanks

@Vatavuk let me know

**Comment by Vatavuk** [comment]
@SharplEr changes are not copy-paste, each class has their own changes. Just by glancing through the commit I see that I have comments that affect almost every file. I don't want to spend more than 15min on this. `ARC` also needs to spend more time on doublechecking. You are always at risk when you submit large PR. If you take a look at other PR you will much smaller PRs. 
This is similar PR #786 (literally only copy paste) that I regret I didn't split in two. I spend way more than 30min in the end. Please commit 5 classes and accompanied tests and for the rest 5 classes create a puzzle.

File: src/test/java/org/cactoos/func/FuncOfTest.java [hunk 1]
```diff
@@ -50,30 +50,6 @@ public void convertsProcIntoFunc() throws Exception {
         );
     }
 
-    @Test
-    public void convertsProcWithNoResultIntoFunc() throws Exception {
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/func/FuncOfTest.java]: @SharplEr leave this test for now until `FuncOf` is refactored

File: src/main/java/org/cactoos/func/BiFuncOf.java [hunk 2]
```diff
@@ -73,14 +73,6 @@ public BiFuncOf(final Func<X, Z> fnc) {
         this((first, second) -> fnc.apply(first));
     }
 
-    /**
-     * Ctor.
-     * @param proc The proc
-     */
-    public BiFuncOf(final Proc<X> proc) {
```
[the remark this answers is not in the document]
Comment (SharplEr) [reply src/main/java/org/cactoos/func/BiFuncOf.java]: @Vatavuk This ctor already exist. See the 89 line in master, or 81 line in my PR.

File: src/test/java/org/cactoos/func/FuncOfTest.java (hunk 1 above)
Comment (SharplEr) [reply src/test/java/org/cactoos/func/FuncOfTest.java]: @Vatavuk Done.

File: src/main/java/org/cactoos/func/BiFuncOf.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vatavuk) [reply src/main/java/org/cactoos/func/BiFuncOf.java]: @SharplEr correct didn't saw it sry

**Comment by Vatavuk** [comment]
@llorllale good to merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @vatavuk[/z](https://www.0crat.com/u/vatavuk)

**Pull request merged**